### PR TITLE
Add Code of Conduct

### PR DESCRIPTION
**Pull request by jsolbrig**
# Reviewer Checklist

PR author: Please ensure you meet all of the below requirements, and check boxes appropriately.

Reviewers: Please confirm all required testing/documentation has been completed prior to approving.

**Remove lines that are not applicable, explain if you select "NO REQUIRED"**

* [ ] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***existing tests*** (explain why not required)
* [ ] Required ***unit tests*** added and pass for new/modified functionality
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [ ] Required ***integration tests*** added and pass for new/modified functionality
* [x] NO REQUIRED ***integration tests*** (explain why not required)
* [x] Required ***documentation*** added for new/modified functionality
* [ ] NO REQUIRED ***documentation*** (explain why not required)
* [x] Required ***release notes*** added for new/modified functionality
* [ ] NO REQUIRED ***release notes*** (explain why not required)
* [ ] Required ***updates to other repos*** complete
* [x] NO REQUIRED ***updates to other repos*** (explain why not required)

https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#604

# Testing Instructions
Just read `CODE_OF_CONDUCT.md` and make sure it seems reasonable for this organization.

# Summary

Add Code of Conduct
-------------------
Borrowed the Contributor Covenant Code of Conduct v2.1, but removed the enforcement section since it doesn't appear to work well for GitHub since it requires private communication. Also added to the documentation using the m2r2 package which was added as a requirement to the `doc` target in pyproject.toml.

::

    added: CODE_OF_CONDUCT.md
    modified: docs/source/new-docs/contribute/code-of-conduct/index.rst
    modified: docs/source/_templates/conf_PKG.py
    modified: pyproject.toml

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official email address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Comment (biosafetylvl5) [inline CODE_OF_CONDUCT.md:64]: need to add link to contact page

**Comment by biosafetylvl5** [comment]
Builds well for me! :)

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (biosafetylvl5) [reply CODE_OF_CONDUCT.md]: Added :)

**Review by biosafetylvl5 — APPROVED** [review]
Thanks @jsolbrig 🙂

**Review by mindyls — APPROVED** [review]
Looks good to me!

**Pull request merged**